README.md

```diff
@@ -1 +1,208 @@
+
+* **Byte Pattern** is a byte sequence used as a template to be matched against in the pattern matching algorithm. 
+* **Pattern Mask** is a byte sequence used to determine the significance of bytes being compared against a byte pattern in the pattern matching algorithm.
+* **Leading Bytes** is set of bytes to be ignored while matching the leading bytes in the content.
+
+---
+**NOTE**
+
+In a pattern mask, `0xFF` indicates the byte is strictly significant, `OxDF` indicates that the byte is significant in an ASCII case-insensitive way, and 0x00 indicates that the byte is not significant. 
+
+---
+
+**Example using `extra_types` argument:**
+```python
+>>> from xtractmime import extract_mime
+>>> extra_types = ((b'test', b'\xff\xff\xff\xff', None, b'text/test'),)
+>>> extract_mime(b'test', extra_types=extra_types)
+b'text/test'
+>>> extract_mime(b"test")
+b'text/plain'
+```
+
+Optional `supported_types` argument is a set of MIME types that are supported by the user agent.
+
+**Example using `supported_types` argument:**
+```python
+>>> from xtractmime import extract_mime
+>>> content_types = (b'image/gif',)
+>>> supported_types = (b'image/gif',)
+>>> extract_mime(b'GIF87a', content_types=content_types, supported_types=supported_types)
+b'image/gif'
+>>> supported_types = (b'audio/mpeg',)
+>>> extract_mime(b'GIF87a', content_types=content_types, supported_types=supported_types)
+b'image/gif'
+```
+
+### function `xtractmime._find_unknown_mimetype : Optional[bytes]`
+**Parameters:**
+
+* `input_bytes: bytes`
+* `sniff_scriptable: bool`
+* `extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]]`
+
+Return a recognized MIME type essence if the format of the content is identified else return a binary MIME type essence. The function is called by `xtractmime.extract_mime` when *Content-Type* headers are unknown (i.e. `content_types` argument is `None`)
+
+### function `xtractmime._sniff_mislabled_binary : Optional[bytes]`
+**Parameters:**
+
+* `input_bytes: bytes`
+
+Return a text or a binary MIME type essence. The function determines whether a binary resource has been mislabeled as plain text.
+The function is related to a bug in some older installations of *Apache* that causes the provided byte sequences to supply
+one of the *Content-Type* headers mentioned in [Section 5.1](https://mimesniff.spec.whatwg.org/commit-snapshots/609a3a3c935fbb805b46cf3d90768d695a1dcff2/#interpreting-the-resource-metadata) of MIME sniffing standards when serving files with unrecognized MIME types.
+
+### function `xtractmime._sniff_mislabled_feed : Optional[bytes]:`
+**Parameters:**
+
+* `input_bytes: bytes`
+* `supplied_type: bytes`
+
+Return a MIME type essence if the content is identified as a XML-based RSS feed or an Atom feed else return supplied MIME type. The function determines whether a feed has been mislabeled as HTML.
+
+### function `xtractmime.is_binary_data : bool`
+**Parameters:**
+
+* `input_bytes: bytes`
```

Comment:
In this case maybe it would be best to keep the parameter in the header, for brevity.